src/storage/store.js

```diff
@@ -360,12 +359,13 @@ export default class Store extends EventTarget {
 
   update(newState, mergeOpts) {
     const finalState = merge(this.state, newState, mergeOpts);
-    const { valid } = validator.validate(finalState, SCHEMA);
+    const { valid, errors } = validator.validate(finalState, SCHEMA);
 
+    // Cleanup unsupported properties
     if (!valid) {
-      // Intentionally not including details about the state or validation result here, since we don't want to leak
-      // sensitive data in the error message.
-      throw new Error(`Write to store failed schema validation.`);
+      errors.forEach(error => {
+        delete error.instance[error.argument];
+      });
```

Comment:
It seems like we should at least log these errors. Also, is this change necessary/desirable given the `allowAdditionalProps` change you referred to in an earlier comment.

Reply:
First comment here https://github.com/mozilla/hubs/pull/4275/files#r636548037 talked about cleaning up the store to avoid stall prefs to stick around forever. I thought it was a good time for adding that.